### ISSUE TITLE
OneStatementPerLine on multiple field initialization

### ISSUE THREAD

**Issue by guillaumecle** [checkstyle/checkstyle#1237]
I don't understand the behavior of OneStatementPerLine on multiple field initialization.
```java
  /**
   * This is detected illegal
   */
  private int
    aField = 0,
    anotherField = 0;

  /**
   * This is detected legal
   */
  private int
    aMember = 0,
    anotherMember = 0
    ;
```
Is it the expected behavior?

**Comment by romani** [comment]
thanks for bug :) report.

**Comment by guillaumecle** [comment]
What is exactly the expected behavior?

```java
  private int
    aField = 0,
    anotherField = 0;
```

Is this legal? In a way it's only one statement per 'file line'.

**Comment by romani** [comment]
>   private int
    aField = 0,
    anotherField = 0;

that code should not raise violation.

FYI: http://checkstyle.sourceforge.net/config_coding.html#MultipleVariableDeclarations should raise there violation (
http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141270.html )
, but is different Check and could be missed in your config.

**Comment by romani** [comment]
merged.

**Comment by guillaumecle** [comment]
Thank to both of you!

**Comment by romani** [comment]
@guillaumecle ,  it would good if you verify on your code that fix before release.

**Comment by guillaumecle** [comment]
I'm not getting any warning anymore. Thank you.

Is there a rule that prevents multiple assignments on one line?
```java
int a = 1, b = 2;
```

**Comment by romani** [comment]
There is no such Check please create an issue on that, it will very tiny and easy to do Check.

**Comment by MEZk** [comment]
@romani 
I've already have an idea how to implement this Check. And, for example we can provide an option "multipleAssignmentsNumber" etc to allow user to config this Check.
